### PR TITLE
[MM-23565] ssh: add start command

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
Added `Start(cmd)` method to run a command w/o waiting it to finish. I can update the PR if others (e.g. `Wait`) are needed.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-23565

### CODE REVIEW

File: deployment/terraform/ssh/ssh.go [hunk 1]
```diff
@@ -81,6 +81,18 @@ func (sshc *Client) RunCommand(cmd string) error {
 	return sess.Run(cmd)
 }
 
+// StartCommand starts a given command in a new ssh session. Unlike RunCommand
+// this command does not wait command to finish.
```
Comment (agnivade) [inline deployment/terraform/ssh/ssh.go]: Please also add this: "This is needed for running commands in the background."

**Comment by streamer45** [comment]
> Added `Start(cmd)` method to run a command w/o waiting it to finish. I can update the PR if others (e.g. `Wait`) are needed.
> 

Does this also mean we never check if it failed or succeeded?

File: deployment/terraform/ssh/ssh.go (hunk 1 above)
Comment (isacikgoz) [reply deployment/terraform/ssh/ssh.go]: done 👍

**Comment by isacikgoz** [comment]
@streamer45 we don't check the result of the command, we only check if it is failed to be executed. My interpretation from the [exec/start](https://pkg.go.dev/os/exec?tab=doc#Cmd.Start) though :)

**Comment by streamer45** [comment]
> @streamer45 we don't check the result of the command, we only check if it is failed to be executed. My interpretation from the [exec/start](https://pkg.go.dev/os/exec?tab=doc#Cmd.Start) though :)

So what kind of command do we plan using this on?
cc @agnivade

**Comment by agnivade** [comment]
Things like starting agents and coordinators. They aren't servicified yet. Hence we need to start them in the background for the program to continue execution.

**Comment by streamer45** [comment]
> Things like starting agents and coordinators. They aren't servicified yet. Hence we need to start them in the background for the program to continue execution.

Oh okay, so daemons that are not expected to exit.

**Pull request merged**